File: src/main/java/szewek/flux/recipe/AlloyingRecipe.java
package szewek.flux.recipe;

import net.minecraft.item.ItemStack;
import net.minecraft.util.ResourceLocation;
import szewek.flux.F;

public final class AlloyingRecipe extends AbstractMachineRecipe {
	public AlloyingRecipe(ResourceLocation idIn, MachineRecipeSerializer.Builder builder) {
		super(F.R.ALLOYING, idIn, builder);
	}

	@Deprecated
	public AlloyingRecipe(ResourceLocation idIn, String group, MachineRecipeSerializer.Builder builder) {
		this(idIn, builder.withGroup(group));
	}

	@Override
	public ItemStack getToastSymbol() {
		return new ItemStack(F.B.ALLOY_CASTER);
	}
}

File: src/main/java/szewek/flux/recipe/AlloyingRecipe.kt
package szewek.flux.recipe

import net.minecraft.item.ItemStack
import net.minecraft.item.crafting.IRecipeSerializer
import net.minecraft.util.ResourceLocation
import szewek.flux.FBlocks
import szewek.flux.FRecipes

class AlloyingRecipe(idIn: ResourceLocation, groupIn: String, builder: MachineRecipeSerializer.Builder)
    : AbstractMachineRecipe(FRecipes.ALLOYING, idIn, groupIn, builder) {
    override fun getIcon() = ItemStack(FBlocks.ALLOY_CASTER)

    override fun getSerializer(): IRecipeSerializer<*> = FRecipes.ALLOYING_SERIALIZER
}